### PR TITLE
Adjustments to WindowPropertyFrameAndDivider and new model gtest

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

While reviewing #4550 I noticed the WindowPropertyFrameAndDivider had no getter/setter model tests, and upon investigation I noticed some issues in the IDD.

The issues are explained in the report I made to EnergyPlus: https://github.com/NREL/EnergyPlus/issues/9344, but the TLDR:
* Frame Conductance currently is neither defaulted not required, yet E+ will silently (in the source code) assume it's 0.0
* Number of Horizontal/Vertical Dividers are marked as `real` but they are actually integers in the source code

This PR does both of these IDD changes and breaks API by making `frameConductance() `return a non-optional double, and the number of dividers take/return int instead of double.

This PR also adds a full Gtest for getters/setters using the upgraded code generation from https://github.com/NREL/OpenStudio/pull/4554

### Pull Request Author

 - [x] Model API Changes / Additions
 - [x] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [x] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [x] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [x] All new and existing tests passes

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: resources/model/OpenStudio.idd [hunk 1]
```diff
@@ -5695,6 +5695,7 @@ OS:WindowProperty:FrameAndDivider,
        \units W/m2-K
        \type real
        \minimum 0.0
+       \default 0.0
```
Comment (jmarrec) [inline resources/model/OpenStudio.idd:5698]: New idd default for Frame Conductance

File: src/model/WindowPropertyFrameAndDivider.hpp [hunk 2]
```diff
@@ -75,7 +75,9 @@ namespace model {
 
     bool isFrameInsideProjectionDefaulted() const;
 
-    boost::optional<double> frameConductance() const;
+    double frameConductance() const;
+
+    bool isFrameConductanceDefaulted() const;
```
Comment (jmarrec) [inline src/model/WindowPropertyFrameAndDivider.hpp:80]: API break on return type for frameConductance() + add a isFrameConductanceDefaulted()

File: src/model/WindowPropertyFrameAndDivider.hpp [hunk 3]
```diff
@@ -101,11 +103,11 @@ namespace model {
 
     bool isDividerWidthDefaulted() const;
 
-    double numberOfHorizontalDividers() const;
+    int numberOfHorizontalDividers() const;
```
Comment (jmarrec) [inline src/model/WindowPropertyFrameAndDivider.hpp:106]: API break on return type for both # of dividers

File: src/model/WindowPropertyFrameAndDivider.hpp [hunk 4]
```diff
@@ -209,11 +211,11 @@ namespace model {
 
     void resetDividerWidth();
 
-    bool setNumberOfHorizontalDividers(double numberOfHorizontalDividers);
+    bool setNumberOfHorizontalDividers(int numberOfHorizontalDividers);
```
Comment (jmarrec) [inline src/model/WindowPropertyFrameAndDivider.hpp:214]: API break on setter argument type for both # of dividers

File: src/model/test/WindowPropertyFrameAndDivider_GTest.cpp [hunk 5]
```diff
@@ -115,3 +115,376 @@ TEST_F(ModelFixture, WindowPropertyFrameAndDivider_Name) {
...
+  // Set
+  EXPECT_TRUE(windowPropertyFrameAndDivider.setFrameWidth(0.75));
+  EXPECT_EQ(0.75, windowPropertyFrameAndDivider.frameWidth());
+  EXPECT_FALSE(windowPropertyFrameAndDivider.isFrameWidthDefaulted());
+  // Bad Value
+  EXPECT_FALSE(windowPropertyFrameAndDivider.setFrameWidth(-10.0));
+  EXPECT_EQ(0.75, windowPropertyFrameAndDivider.frameWidth());
+  // Reset
+  windowPropertyFrameAndDivider.resetFrameWidth();
+  EXPECT_TRUE(windowPropertyFrameAndDivider.isFrameWidthDefaulted());
```
Comment (jmarrec) [inline src/model/test/WindowPropertyFrameAndDivider_GTest.cpp:137]: New full getters/setters GTest

**Comment by tijcolem** [comment]
@jmarrec Looks good! Do you mind checking on windows to see why the build failed?

File: src/energyplus/ReverseTranslator/ReverseTranslateWindowPropertyFrameAndDivider.cpp [hunk 6]
```diff
@@ -104,14 +104,14 @@ namespace energyplus {
...
-      result.setNumberOfHorizontalDividers(*d);
+    boost::optional<int> i = workspaceObject.getInt(WindowProperty_FrameAndDividerFields::NumberofHorizontalDividers);
+    if (i) {
+      result.setNumberOfHorizontalDividers(*i);
     }
 
-    d = workspaceObject.getDouble(WindowProperty_FrameAndDividerFields::NumberofVerticalDividers);
-    if (d) {
-      result.setNumberOfVerticalDividers(*d);
+    i = workspaceObject.getDouble(WindowProperty_FrameAndDividerFields::NumberofVerticalDividers);
```
Comment (jmarrec) [inline src/energyplus/ReverseTranslator/ReverseTranslateWindowPropertyFrameAndDivider.cpp:112]:
```suggestion
    i = workspaceObject.getInt(WindowProperty_FrameAndDividerFields::NumberofVerticalDividers);
```

@tijcolem this is the mistake I made that Windows is throwing on (which is actually helpful for once)

Comment (jmarrec) [reply src/energyplus/ReverseTranslator/ReverseTranslateWindowPropertyFrameAndDivider.cpp]: Resolved in 81308adf2483eb6877d00d9e3c471f4701184d18 (after rebasing on develop)

**Pull request merged**